Logger/NameRecords.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Xml.Linq;
+
+namespace Logger
+{
+    public readonly record struct NameRecords(string firstName, string lastName, string? middleName = null)
```

Comment:
```suggestion
    public readonly record struct Name(string firstName, string lastName, string? middleName = null)
```
Typically, you wouldn't name a record with the term `Record` included in the name, also, class names should be singular.